src/python/gudhi/simplex_tree_multi.pyx

```diff
@@ -0,0 +1,970 @@
+		"""
+		#TODO : check
+		low = np.min([f for s,F in self.get_simplices() for f in np.array_split(F, len(F) // self.num_parameters)], axis=0)
+		high = np.max([f for s,F in self.get_simplices() for f in np.array_split(F, len(F) // self.num_parameters)], axis=0)
+		return np.asarray([low,high])
+
+	
+
+	def fill_lowerstar(self, F, parameter:int):
+		""" Fills the `dimension`th filtration by the lower-star filtration defined by F.
+
+		Parameters
+		----------
+		F:1d array
+			The density over the vertices, that induces a lowerstar filtration.
+		parameter:int
+			Which filtration parameter to fill. /!\ python starts at 0.
+
+		Returns
+		-------
+		self:SimplexTree
+		"""
+		# for s, sf in self.get_simplices():
+		# 	self.assign_filtration(s, [f if i != dimension else np.max(np.array(F)[s]) for i,f in enumerate(sf)])
+		self.get_ptr().fill_lowerstar(F, parameter)
+		return self
+
+
+	def project_on_line(self, parameter:int=0, basepoint:None|list|np.ndarray= None):
+		"""Converts an multi simplextree to a gudhi simplextree.
+		Parameters
+		----------
+			parameter:int = 0
+				The parameter to keep. WARNING will crash if the multi simplextree is not well filled.
+			basepoint:None
+				Instead of keeping a single parameter, will consider the filtration defined by the diagonal line crossing the basepoint.
+		WARNING 
+		-------
+			There are no safeguard yet, it WILL crash if asking for a parameter that is not filled.
+		Returns
+		-------
+			A gudhi simplextree with chosen 1D filtration.
+		"""
+		# FIXME : deal with multicritical filtrations
+		import gudhi as gd
+		new_simplextree = gd.SimplexTree()
+		assert parameter < self.get_ptr().get_number_of_parameters()
+		cdef int c_parameter = parameter
+		cdef intptr_t old_ptr = self.thisptr
+		cdef intptr_t new_ptr = new_simplextree.thisptr
+		cdef vector[value_type] c_basepoint = [] if basepoint is None else basepoint
+		if basepoint is None:
+			with nogil:
+				flatten(old_ptr, new_ptr, c_parameter)
+		else:
+			with nogil:
+				flatten_diag(old_ptr, new_ptr, c_basepoint, c_parameter)
+		return new_simplextree
+
+	def set_num_parameter(self, num:int):
+		"""
+		Sets the numbers of parameters. 
+		WARNING : it will resize all the filtrations to this size. 
+		"""
+		self.get_ptr().resize_all_filtrations(num)
+		self.get_ptr().set_number_of_parameters(num)
+		return
+
+	def __eq__(self, other:SimplexTreeMulti):
+		"""Test for structural equality
+		:returns: True if the 2 simplex trees are equal, False otherwise.
+		:rtype: bool
+		"""
+		return dereference(self.get_ptr()) == dereference(other.get_ptr())
+	
+	def euler_char(self, points:np.ndarray) -> np.ndarray:
+		""" Computes the Euler Characteristic of the filtered complex at given (multiparameter) time
+
+		Parameters
+		----------
+		points: 2-dimensional array.
+			List of filtration values on which to compute the euler characteristic.
+			WARNING FIXME : the points have to have the same dimension as the simplextree.
```

Comment:
There are several notions of dimension for a simplex tree, I think you mean the one of the filtration values.

Reply:
Yes, the size of the points have to be of the same size as `self.num_parameters`